ejb/src/main/java/org/wildfly/httpclient/ejb/EjbHttpService.java

```diff
@@ -95,11 +89,13 @@ private void registerHandler(final PathHandler pathHandler, final RequestType re
     }
 
     private HttpHandler newInvocationHandler(final RequestType requestType) {
-        if (requestType == CANCEL) return new HttpCancelHandler(association, executorService, localTransactionContext, cancellationFlags);
-        if (requestType == CREATE_SESSION) return new HttpSessionOpenHandler(association, executorService, localTransactionContext, httpServiceConfig);
-        if (requestType == DISCOVER) return new HttpDiscoveryHandler(executorService, association, httpServiceConfig);
-        if (requestType == INVOKE) return new HttpInvocationHandler(association, executorService, localTransactionContext, cancellationFlags, classResolverFilter, httpServiceConfig);
-        throw new IllegalStateException();
+        switch (requestType) {
```

Comment:
I generally prefer including the code block on the line following the case identifier (with the appropriate indentation):
case INVOKE:
    return new ...
case CANCEL:
    return new ...

Reply:
Will keep that in mind.